waspc/cli/src/Wasp/Cli/Command/Db/Start.hs

```diff
@@ -0,0 +1,129 @@
+import Wasp.Project.Db (databaseUrlEnvVarName)
+import Wasp.Project.Db.Dev (makeDevDbUniqueId)
+import qualified Wasp.Project.Db.Dev.Postgres as Dev.Postgres
+import Wasp.Project.Env (dotEnvServer)
+
+-- | Starts a "managed" dev database, where "managed" means that
+-- Wasp creates it and connects the Wasp app with it.
+-- Wasp is smart while doing this so it checks which database is specified
+-- in Wasp configuration and spins up a database of appropriate type.
+start :: Command ()
+start = do
+  waspProjectDir <- findWaspProjectRootDirFromCwd
+  appSpec <- analyze waspProjectDir
+
+  throwIfThereIsCustomDbUrl appSpec
+
+  let (appName, app) = ASV.getApp appSpec
+  case getDbSystem app of
+    AS.App.Db.SQLite -> noteSQLiteDoesntNeedStart
+    AS.App.Db.PostgreSQL -> startPostgreDevDb waspProjectDir appName
+  where
+    noteSQLiteDoesntNeedStart =
+      cliSendMessageC . Msg.Info $
+        "Nothing to do! You are all good, you are using SQLite which doesn't need to be started."
+
+    throwIfThereIsCustomDbUrl :: AS.AppSpec -> Command ()
+    throwIfThereIsCustomDbUrl spec =
+      when (isThereCustomDbUrl spec) $
+        E.throwError $
+          CommandError
+            "You are using custom database already"
+            ( printf
+                ( "Wasp has detected that you have defined %s env var in your %s file.\n"
+                    <> "To have Wasp run the database for you, make sure you remove that env var first."
+                )
+                databaseUrlEnvVarName
+                (fromRelFile (dotEnvServer :: Path' (Rel WaspProjectDir) File'))
+            )
+
+    isThereCustomDbUrl :: AS.AppSpec -> Bool
+    isThereCustomDbUrl = any ((== databaseUrlEnvVarName) . fst) . AS.devEnvVarsServer
+
+getDbSystem :: AS.App.App -> AS.App.Db.DbSystem
+getDbSystem app =
+  fromMaybe AS.App.Db.SQLite (AS.App.db app >>= AS.App.Db.system)
+
+startPostgreDevDb :: Path' Abs (Dir WaspProjectDir) -> String -> Command ()
+startPostgreDevDb waspProjectDir appName = do
+  throwIfExeIsNotAvailable
+    "docker"
+    "To run PostgreSQL dev database, Wasp needs `docker` installed and in PATH."
+
+  cliSendMessageC . Msg.Info $
+    unlines
+      [ "✨ Starting dev database for your Wasp app (detected PostgreSQL) ✨",
+        "",
+        "Additional info:",
+        " ℹ Connection URL, in case you might want to connect with external tools:",
+        "     " <> connectionUrl,
+        " ℹ Database data is persisted in a docker volume with the following name"
+          <> " (useful to know if you will want to delete it at some point):",
+        "     " <> dockerVolumeName
+      ]
+
+  -- NOTE: POSTGRES_PASSWORD, POSTGRES_USER, POSTGRES_DB below are really used by the docker image
+  --   only when initializing the database -> if it already exists, they will be ignored.
+  --   This is how the postgres Docker image works.
+  let command =
+        printf
+          ( unwords
+              [ "docker run",
```

Comment:
Suggestion: we could add the `--name` param as well, so it looks nicer in the list of containers. We could use the same name as the db name 🤷

<img width="596" alt="Screenshot 2023-03-13 at 12 16 07" src="https://user-images.githubusercontent.com/2223680/224686482-68018599-ea0b-4e0d-b9df-9f702301f656.png">

Reply:
Good idea, I did it!